pkgs/development/compilers/flutter/update.py

```diff
@@ -0,0 +1,359 @@
+          flutterPlatforms = [
+            "android"
+            "ios"
+            "web"
+            "linux"
+            "windows"
+            "macos"
+            "fuchsia"
+            "universal"
+          ];
+          systemPlatforms = [
+            "x86_64-linux"
+            "aarch64-linux"
+            "x86_64-darwin"
+          ];
+
+          derivations =
+            lib.foldl' (
+              acc: flutterPlatform:
+                acc
+                ++ (map (systemPlatform:
+                  callPackage "%s/pkgs/development/compilers/flutter/artifacts/fetch-artifacts.nix" {
+                    flutter = flutter%s;
+                    inherit flutterPlatform;
+                    inherit systemPlatform;
+                    hash = "";
+                  })
+                systemPlatforms)
+            ) []
+            flutterPlatforms;
+        in
+          symlinkJoin {
+            name = "evaluate-derivations";
+            paths = derivations;
+          }
+    """.strip() % (flutter_compact_version, NIXPKGS_ROOT, flutter_compact_version)
+
+    stderr = nix_build_to_fail(code)
+
+    pattern = re.compile(
+        r"/nix/store/.*-flutter-artifacts-(.+?)-(.+?).drv':\n\s+specified: .*\n\s+got:\s+(.+?)\n")
+    matches = pattern.findall(stderr)
+    result_dict = {}
+
+    for match in matches:
+        flutter_platform, architecture,  got = match
+        result_dict.setdefault(flutter_platform, {})[architecture] = got
+
+    def sort_dict_recursive(d):
+        return {k: sort_dict_recursive(v) if isinstance(v, dict) else v for k, v in sorted(d.items())}
+    result_dict = sort_dict_recursive(result_dict)
+
+    return object_to_nix(result_dict)
+
+
+def get_dart_hashes(dartVersion):
+    code = {
+        "x86_64-linux": """
+            { fetchzip }: fetchzip {
+              url = "https://storage.googleapis.com/dart-archive/channels/stable/release/%s/sdk/dartsdk-linux-x64-release.zip";
+              sha256 = "";
+            }
+            """.strip(),
+        "aarch64-linux": """
+            { fetchzip }: fetchzip {
+              url = "https://storage.googleapis.com/dart-archive/channels/stable/release/%s/sdk/dartsdk-linux-arm64-release.zip";
+              sha256 = "";
+            }
+        """.strip(),
+        "x86_64-darwin": """
+            { fetchzip }: fetchzip {
+              url = "https://storage.googleapis.com/dart-archive/channels/stable/release/%s/sdk/dartsdk-macos-x64-release.zip";
+              sha256 = "";
+            }
+        """.strip(),
+        "aarch64-darwin": """
+            { fetchzip }: fetchzip {
+              url = "https://storage.googleapis.com/dart-archive/channels/stable/release/%s/sdk/dartsdk-macos-arm64-release.zip";
+              sha256 = "";
+            }
+        """.strip(),
+    }
+    result_dict = {}
+    for (platform, code) in code.items():
+        stderr = nix_build_to_fail(code % (dartVersion))
+
+        pattern = re.compile(r"got:\s+(.+?)\n")
```

Comment:
You can replicate Nix hashes using the Python stdlib

Example: https://github.com/lucasew/playground/blob/master/piton/fetch-sri-sha256.py

Note that it only works with single files.

Reply:
that doesn't work for me with fetchzip. gives a different hash.

How necessary is that?

Reply:
is fetchzip being run with unpack flag? then it won't replicate them 1:1